### ISSUE TITLE
Documentation: explain how to export smooth/sharp shading (actually, maybe bug or regression?)

### ISSUE THREAD

**Issue by Vinc3r** [KhronosGroup/glTF-Blender-IO#911]
On Blender 2.79, users was allowed to use autosmooth to export smooth & sharp shading in gltf files (which was very convenient by the way, should be a great default behaviour to remerge IMO), but [that's not anymore the case since 2.8 series](https://user-images.githubusercontent.com/3873421/73638928-9d234200-466b-11ea-8931-b5722c0844f9.jpg): user now must use Edge Split modifier and check "apply modifier" option on export (in large scenes this extend the export time, that's why autosmooth option could be great also).

In short, the documentation should mention the workflow to save mesh normals.

**Comment by emackey** [comment]
I don't think this was intentionally turned off.  It sounds like a bug, or a regression.

**Comment by Vinc3r** [comment]
If it's possible to make this functionnality comes back, I'm all in!

**Comment by julienduroure** [comment]
Maybe linked to #877, and child tickets

**Comment by emackey** [comment]
Looks like I'm wrong, this is not a regression.  It's a bug that somehow I thought was fixed that never seems to have been fixed: #799

One workaround (for non-skinned models at least) is to check "Apply Modifiers".  There's an implicit "Edge Split" modifier that is temporarily applied during glTF export when this is checked, and that's what causes the correct auto smooth effects to happen.

Another workaround is to click the button to "Add Custom Split Normal Data".  I haven't tested this much but I think it essentially bakes the results of auto-smoothing into the custom normals, and the exporter uses those.  Should work with skinned models, but might need to be redone (remove and re-add custom normals) after geometry changes (unless Blender does this already, I didn't test it).

**Comment by Vinc3r** [comment]
'was'nt aware about the implicit edge split modifier, great to know (but as I said on my first message, on large scene it's an export-time killer).

About the Add Custom Split Normal Data from what I've tested for now it works. I will add this functionnality to my [ReTiCo](https://github.com/Vinc3r/ReTiCo#retico) addon since user can't apply this operation on multiple selected meshes (using the alt + click trick).

**Comment by julienduroure** [comment]
Closing this ticket, as #877 will remain the main ticket about normal export refactoring
